### PR TITLE
Issue/2086 - Refactored with Bulk Delete

### PR DESCRIPTION
**Pull request by mehul0810**
## Description
This PR is for #2086 

## How Has This Been Tested?
I've tested this PR with multiple scenario's.
- No Donor Selected and trying delete via bulk actions.
- Single Donor Selected and trying delete via bulk actions.
- Multiple Donors Selected and trying delete via bulk actions.
- Clicking Select All Checkbox to select all donors and trying delete via bulk actions.
- Using Delete button from each donor triggering Bulk Delete feature to delete single donor as well. ( i.e. Single Delete Donor, no longer uses additional page )

## Screenshots (jpeg or gifs if applicable):
gif

## Types of changes
New feature (non-breaking change which adds functionality)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

### CODE REVIEW

File: assets/js/admin/admin-scripts.js [hunk 1]
```diff
@@ -1327,23 +1328,88 @@ var give_setting_edit = false;
...
+			$body.on( 'click', '#cb-select-all-1, #cb-select-all-2', function() {
+				$( '#give-donors-filter tbody input[type="checkbox"]' ).trigger( 'click' );
+			});
+
+			// On checking checkbox, add to bulk delete donor.
+			$body.on( 'click', '#give-donors-filter tbody .check-column > input[type="checkbox"]', function() {
+				var donorId   = $( this ).closest( 'tr' ).data( 'id' ),
+					donorName = $( this ).closest( 'tr' ).data( 'name' ),
+					donorHtml = '<div id="give-donor-' + donorId + '">' +
+					'<a data-id="' + donorId + '" class="give-skip-donor" title="Remove From Bulk Delete">X</a>' +
```
Comment (ravinderk) [inline assets/js/admin/admin-scripts.js]: @mehul0810 This string should be internationalized.

File: includes/admin/donors/class-donor-table.php [hunk 2]
```diff
@@ -303,27 +303,22 @@ public function process_bulk_action() {
...
-
-			} // End switch().
+			case 'delete':
+				$args = array(
+					'_wpnonce'  => wp_create_nonce( 'delete-bulk-donors' ),
+					'donor_ids' => $ids,
+				);
+				$message = give_delete_donor( $args );
+				$redirect_url = add_query_arg( 'give-message', $message, admin_url( 'edit.php?post_type=give_forms&page=give-donors' ) );
+//wp_safe_redirect( esc_url($redirect_url) );
```
Comment (ravinderk) [inline includes/admin/donors/class-donor-table.php]: @mehul0810  Remove commented code.

**Comment by mehul0810** [comment]
@ravinderk I've updated code as per our last discussion over call.
1. Fixed bug in select all donor, un check any donor then select all again functioning ( Tried all the possible cases)
2. Fixed Donor Delete confirmation checkbox getting checked when Selected all donors at once.
3. Smoothen the Cancel Bulk Delete option to remove the delay response.

**Comment by mehul0810** [comment]
@ravinderk Also, fixed the issue when there is no name for a donor under bulk delete screen. Please review with latest pull.

**Comment by ravinderk** [comment]
@mehul0810 As we discuss overcall cancel feature is not working per donor when bulk delete tab opened.

jaotcsiozg

**Comment by mehul0810** [comment]
@ravinderk I've fixed and tested the issue.

gif

**Comment by ravinderk** [comment]
@mehul0810 Bulk delete feature is not working with the recurring addon. Donation deleted but not the subscription. I think recurring must have support for the bulk delete donor.

screen shot 2017-11-16 at 10 15 55 am



@mehul0810  Can you test deleting donor from donor edit screen also?

**Comment by mehul0810** [comment]
@ravinderk As discussed over call, I've fixed the positioning of Donor issue on bulk delete and the above issue is already fixed with latest release of `Recurring 1.4`.

**Pull request merged**